util/taskcluster.py

```diff
@@ -58,6 +59,21 @@ def report_progress(count, block_size, total_size):
     else:
         print('File already exists: %s' % target_file)
 
+    try:
+        with gzip.open(target_file, "rb") as f:
+            f.read()
```

Comment:
Instead of opening and reading you could just check the headers returned by `urlretrieve`:

```python
    if not os.path.isfile(target_file):
        print('Downloading %s ...' % tc_url)
        _, headers = urllib.request.urlretrieve(tc_url, target_file, reporthook=(report_progress if progress else None))
        is_gzip = headers.get('Content-Encoding') == 'gzip'
    else:
        print('File already exists: %s' % target_file)
        is_gzip = False
```

Reply:
Have you verified that ? Because when using `wget` / `curl`, `Content-Type` is not `gzip` but `application/octet-stream`.

Reply:
Content-Encoding, not Type.

Reply:
If you prefer keeping the try-except, I'd maybe move the except block up to make it more clear what the purpose of the code is:

```python
    try:
        with gzip.open(target_file, "rb") as f:
            f.read()
        is_gzip = True
    except OSError as ex:
        if "Not a gzipped" in ex.args[0]:
            is_gzip = False
        else:
            raise

    if is_gzip:
        # ...
```